flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/restart/FixedDelayRestartStrategy.java

```diff
@@ -60,9 +63,13 @@ public boolean canRestart() {
 	}
 
 	@Override
-	public void restart(final RestartCallback restarter, ScheduledExecutor executor) {
+	public CompletableFuture<Void> restart(final RestartCallback restarter, ScheduledExecutor executor) {
 		currentRestartAttempt++;
-		executor.schedule(restarter::triggerFullRecovery, delayBetweenRestartAttempts, TimeUnit.MILLISECONDS);
+		Supplier<Void> restartSupplier = () -> {
```

Comment:
This conversion to `Supplier` is duplicated multiple times. See my other comment in `FutureUtils`.